File: c++/66_Plus_One.cpp
#include <iostream>
#include <vector>
#include <map>
#include <string>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        int carry = 1;
        for (int i = n - 1; i >= 0; i--) {
            int sum = digits[i] + carry;
            carry = sum / 10;
            digits[i] = sum % 10;
        }
        if (carry)
            digits.insert(digits.begin(), 1);
        return digits;
        
    }
};


int main() {



}

File: python/66_Plus_One.py
#!/usr/bin/python
import sys
import os

class Solution(object):
    def plusOne(self, digits):
    	carry = 1
    	for i in range(len(digits) - 1, -1, -1):
    		sums = carry + digits[i]
    		carry = sums / 10
    		digits[i] = sums % 10
    	if carry:
    		digits = [1] + digits
    	return digits


def main():
    s = Solution()

if __name__ == "__main__":
    main()
